### ISSUE TITLE
Scope should be larger

### ISSUE THREAD

**Issue by PeterWone** [microsoft/vscode-l10n#21]
While it's great that you are looking at providing a formalised solution for localisation of extension strings this is not all that hard once you get past the total lack of documentation. What's _hard_ is coordinating incremental updates when you have a contributor per translation. For example, I do English and French, and I have colleagues who speak English and Spanish, or English and Italian, or English and Icelandic. Every time I release a minor version, I have to coordinate changes to 

* readme
* changelog
* UI (settings descriptions etc)
* UI dialog messages
* manual

I can almost hear you drawing breath to say that github does diffs.

It's not good enough. 
* It's spread all over the place. 
* There's no coherent way to see what languages have been updated and which haven't.

What's required is tooling to _manage_ localisation.

* Tooling to identify translation candidates (literal strings) 
  * ignore replacement tokens because these are already being localised
  * ignore literals found in an ignore list 
* Tooling to emit difference files requiring translation as a single file for the translator
* Tooling to update the NLS source files from the translated difference file.

If you ask me this begs for implementation as a VS Code extension.

What are your thoughts?

**Comment by TylerLeonhardt** [comment]
The beauty of building a CLI that also had code entrypoints is that we _can_ do both a CLI & a VS Code extension to assist folks. Not entirely sure what an extension would do to help besides be a Command Palette wrapper around the CLI

**Comment by PeterWone** [comment]
I had a notion of a split view of source and target language for the translator. It's not central.

**Comment by TylerLeonhardt** [comment]
Yeah that sounds really nice. If it were a web extension, I like the idea of pointing someone to a vscode.dev/github/.... url where they can contribute without cloning the repo locally

**Comment by PeterWone** [comment]
Another idea was to sketch out the translations using Azure translate so that translators just skim and take a hand when the sense is wrong or particular idiom is required as is often the case for software.

**Comment by TylerLeonhardt** [comment]
> sketch out the translations using Azure translate

This sounds awesome.

**Comment by PeterWone** [comment]
Tyler, sorry to vanish off the face of the Earth... having a day job will do that. Especially if you _also_ have a rural property to run. 

GitHub retired node12 which forces a cascade of updates... and caused me to rediscover this package. Congratulations on the implementation of the Azure Translate idea, it's very nice. 

You may be interested in translate.mjs which is a little js app I put together for my release workflow. It uses `marked` to parse Markdown documents into structural elements so it can present them to Azure Translate and then reconstruct the document. You can't just stuff a Markdown string through Azure Translate, it borks the document in its punctuation translation pass, eg for French it converts all backticks to apostrophe, and then System.Json.Text (which it uses) converts all the apostrophes to `\u0027`. 

If you want to have a look, it's [here](https://github.com/PDConSec/vsc-print/blob/fix/tech-debt/translate.mjs) at the moment but will end up in the root of master.
